capstone/src/main/java/com/google/sps/data/Classroom.java

```diff
@@ -1,94 +1,100 @@
 package com.google.sps.data;
-import java.io.*;
-import com.google.sps.data.Teacher;
-import com.google.sps.data.Student;
-import com.google.sps.data.Form;
+
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Entity;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Random; 
-import com.google.appengine.api.datastore.Entity;
+import com.google.sps.service.DatabaseService;
 
 public class Classroom {
+    protected static final String CLASSROOM = "Classroom";
+    protected static final String TEACHER = "teacher";
+    protected static final String SUBJECT = "subject";
+    protected static final String STUDENTS = "students";
+    protected static final String FORMS = "forms";
 
-    private Teacher teacher;
-    private List<Student> students = new ArrayList<>();
-    private List<Form> forms = new ArrayList<>();
-    private String subject;
-    private Key key;
-
-    public Classroom(Teacher teacher, List<Student> students, List<Form> forms, String subject, Key key) {
-        this.teacher = teacher;
-        this.students = students;
-        this.forms = forms;
-        this.subject = subject;
-        this.key = key;
-    }
+    protected Entity entity;
 
-    // Setters
-    public void setTeacher(Teacher teacher) {
-        this.teacher = teacher;
+    public Classroom(Entity entity) {
+        this.entity = entity;
     }
 
-    public void setSubject(String subject) {
-        this.subject = subject;
+    public Classroom(User teacher, String subject) { 
+        this.entity = new Entity(CLASSROOM);
+        this.entity.setProperty(TEACHER, teacher.getUserKey());
+        this.entity.setProperty(SUBJECT, subject);
+        this.entity.setProperty(STUDENTS, new ArrayList<>());
+        this.entity.setProperty(FORMS, new ArrayList<>());
     }
 
-    public void setKey(Key key) {
-        this.key = key;
+    public void setTeacher(User teacher) { 
+        this.entity.setProperty(TEACHER, teacher.getUserKey());
     }
-
-    // Getters
-    public Teacher getTeacher() {
-        return this.teacher;
+    
+    public Key getTeacher() {
+        return (Key) this.entity.getProperty(TEACHER);
     }
 
     public String getSubject() {
-        return this.subject;
+        return (String) this.entity.getProperty(SUBJECT);
```

Comment:
I'm not sure about the `Entity` API, but you _may_ need to check for the null case?

Reply:
are you saying in the off chance that someone tries to create an object and inputs null values as the parameters?

Reply:
As I had said in another standup: never assume anything about the chances of things. Your code should operate on as few assumptions as possible, including not assuming the data in the database is good data produced in the currently expected way.

That said, I dont think the null case needs to be checked since (horray... Java 😒 ) null is a valid value for all objects.

Reply:
we can talk about this more in the standup about how to handle null inputs

Reply:
resolved: will add in a future PR